File: leetcode/125_easy_valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int i=0,j=s.size()-1;
        while(i<=j){
            // cout<<i<<" "<<j<<" "<<s[i]<<" "<<s[j]<<endl;
            if(isalpha(s[i]) && isalpha(s[j])){
                if(tolower(s[i])==tolower(s[j])){
                    i++;
                    j--;
                }
                else 
                    return false;
            }
            else if(isdigit(s[i]) && isdigit(s[j])){
                if(s[i]==s[j]){
                    i++;
                    j--;
                }
                else 
                    return false;
            }
            else if((isdigit(s[i]) && isalpha(s[j]) ) || (isdigit(s[j]) && isalpha(s[i]) ))
                    return false;
            else if(!(isalpha(s[i]) || isdigit(s[i])))
                i++;
            else if(!(isalpha(s[j]) || isdigit(s[j])))
                j--;
        }
       return true; 
    }
};

File: leetcode/125_easy_valid-palindrome.py
class Solution(object):
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        forward_ptr = 0
        backward_ptr = len(s) - 1

        while forward_ptr < backward_ptr:
            while (forward_ptr < backward_ptr) and not s[forward_ptr].isalnum():
                forward_ptr += 1
            while (forward_ptr < backward_ptr) and not s[backward_ptr].isalnum():
                backward_ptr -= 1
            if s[forward_ptr].lower() != s[backward_ptr].lower():
                return False
            forward_ptr += 1
            backward_ptr -= 1
        return True